lib/notice.js

```diff
@@ -0,0 +1,57 @@
+const { MessageEmbed } = require('discord.js')
+const config = require('./config')
+
+const userCooldowns = new Map()
+const channelCooldowns = new Map(config.channels.notices.map(v => [v, 0]))
+
+function cleanup () {
+  const now = Date.now()
+  for (const [user, timeout] of userCooldowns) {
+    if (timeout < now) {
+      userCooldowns.delete(user)
+    }
+  }
+}
+
+function isEnabled () {
+  return this.enabled && (this.embed.title || this.embed.description)
+}
+
+function processMessage (msg) {
+  if (!this.isEnabled()) {
+    return
+  }
+
+  if (msg.member.roles.highest.name !== '@everyone') {
```

Comment:
I think it would be better to use discord api here (e.g msg.guild.roles.everyone) instead of hardcoding it

Reply:
Additionally there should be guild check here so the messages wont get processed in bot DMs